crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -1173,6 +1294,21 @@ impl<'db> BindingError<'db> {
                     ),
                 );
             }
+
+            Self::InternalCallError(reason) => {
+                context.report_lint(
+                    &CALL_NON_CALLABLE,
```

Comment:
This is probably not ideal. Should I introduce a new diagnostic for this?

Reply:
I wouldn't bother adding a new diagnostic kind for this. I think what you have here is fine for this PR. I suspect that in future with sub-diagnostics we will want to do something a bit more sophisticated here, where instead of just having an opaque string we can store the binding from the internal call.